### PR TITLE
Updated to allow for key padding

### PR DESCRIPTION
**Pull request by mnkolev**
We ran into an intermittent issue where the session key generated isn't always 2048b. Normally this isn't an issue but since we store our keys in the Luna CloudHSM, it expects an exact key size and thus rejects the request.

This minor code fix adds padding where needed for all key sizes.

### CODE REVIEW

**Comment by alonbl** [comment]
Why a new pull request?
Please use the same [previous] branch and push with --force

**Comment by alonbl** [comment]
Please? just rename the branch name to your old branch name and push with --force

**Comment by mnkolev** [comment]
I had to create a new fork and PR based on a request that came in from open source contribution guidelines from above. Would we be ok with merging in this PR?

**Comment by alonbl** [comment]
You must work with git correctly, even for the review iterations... take it as opportunity to learn...

You should not have created a new fork... but even if you have, you can use the same branch, oh... I see you even deleted the previous repository... ok... very difficult to work this way

File: gnupg-pkcs11-scd/command.c [hunk 1]
```diff
@@ -1324,18 +1324,39 @@ gpg_error_t cmd_pkdecrypt (assuan_context_t ctx, char *line)
...
+	{
 		_data.data++;
 		_data.size--;
 	}
+	else
+		fixuplen = 0;
+
+	if (fixuplen > 0)
+	{
+		newdata = malloc(fixuplen + _data.size);
```
Comment (alonbl) [inline gnupg-pkcs11-scd/command.c:1354]: where do you free this?

**Comment by mnkolev** [comment]
Correct, I had everything saved locally but was requested to do so and it made things a bit more difficult, I apologize.

File: gnupg-pkcs11-scd/command.c [hunk 2]
```diff
@@ -1324,18 +1324,39 @@ gpg_error_t cmd_pkdecrypt (assuan_context_t ctx, char *line)
...
 			_data.size == 193 ||
 			_data.size == 257 ||
 			_data.size == 385 ||
-			_data.size == 513
-		)
-	) {
+			_data.size == 513)) 
+	{
 		_data.data++;
 		_data.size--;
```
Comment (alonbl) [inline gnupg-pkcs11-scd/command.c:1347]: we repeat these numbers over and over, we need to optimize...

something like:

```c
static const int keysizes[] = {1024/8, 1536/8, ...., 0};
char *_dataptr = NULL;
int *s;
for (s = keysizes; *s != 0; s++) {
    if (_data.size >= *s - 16 && _data.size < *s) {
        int fixuplen = *s - _data.size;
        dataptr = calloc(fixuplen + _data.size);
        memcpy(dataptr + fixuplen, _data.data, _data.size);
        _data.data = dataptr;
        _data.size += fixuplen;
        break;
    }
    if (*_data.data == 0 && _data.size == *s+1) {
        _data.data++;
        _data.size--;
       break;
    }
}

...

cleanup:
if (dataptr != NULL) {
    free(dataptr);
    dataptr = NULL;
}
```

**Comment by mnkolev** [comment]
Waiting on another approval from open source team, will make these changes in the mean time. Appreciate you looking over this

**Comment by alonbl** [comment]
Probably a bug in the HSM

**Pull request closed without merging**